crates/ruma-events/src/room/name.rs

```diff
@@ -14,7 +16,7 @@ pub type NameEvent = StateEvent<NameEventContent>;
 pub struct NameEventContent {
     /// The name of the room. This MUST NOT exceed 255 bytes.
     #[serde(default, deserialize_with = "room_name")]
-    pub(crate) name: Option<String>,
+    pub(crate) name: Option<RoomName>,
```

Comment:
With the type now enforcing the character limitation, the field can be made public and the `MUST NOT exceed` part of the docs removed.